### ISSUE TITLE
`teardownContext` causing tests to hang

### ISSUE THREAD

**Issue by adriancooney** [emberjs/ember-test-helpers#528]
`teardownContext` is causing my tests to hang. I have an AJAX request that I expected to hit an endpoint but due to a programmer error in my test, I forgot to `await` the method that would hit the endpoint. This mean't the test ran to completion immediately and but the teardown would hang. After further investigation, I found the teardown could *never* complete because we remove the mechanism for updating the settled state values for pending requests.

https://github.com/emberjs/ember-test-helpers/blob/4ad756ee62ee6c87b1a9ec0e27a66bc5c3b092e7/addon-test-support/%40ember/test-helpers/teardown-context.ts#L36-L45

We call `_teardownAJAXHooks` which removes the listeners that will mutate `pendingRequests` and then we call `settled` which waits on the value of `pendingRequests` to change to zero. As a consequence of this, the tests hang until timeout.

**Comment by rwjblue** [comment]
Ya, definitely a bug. We can do one of the following (open to other suggestions too though):

* reset pendingRequests to 0 when we teardown the listeners
* wait for settled before tearing them down
* we use a flag to indicate that AJAX requests should be pushed onto `pendingRequests` or not, instead of _actually_ setting up the `ajaxStart` / `ajaxComplete` listeners for each test. This would allow existing `pendingRequests` to be cleared (and therefore prevent the indefinite unsettled state), prevent _new_ requests from being tracked until the next test starts, and make each test have a small amount of less work overall...

I vaguely prefer the last option.

@adriancooney / @Turbo87 - what do you think?

**Comment by Turbo87** [comment]
>     * wait for settled before tearing them down

seems like a good option to me 🤔

**Comment by rwjblue** [comment]
@Turbo87 - ya, it could work but it has the following problems:

* requires _two_ `await settled()`'s in `teardownContext`
* doesn't have a path forward in the world where ember-qunit's `setupTestIsolationValidation` feature is enabled (and therefore we don't `await settled()` at all)

Frankly, that last point _might_ suggest that we don't have a problem anyways. In @adriancooney's case with `start({ setupTestIsolationValidation: true })` in `tests/test-helper.js` this exact test would have errored (due to async leakage) which is correct AFAICT...

**Comment by yads** [comment]
I've opened a new PR, that attempts to mimic ember-qunit in ember-mocha https://github.com/emberjs/ember-mocha/pull/522
